src/lib/utils/hooks/SynapseAPI/oauth/useOAuthClient.ts

```diff
@@ -19,3 +30,67 @@ export function useGetOAuthClientInfinite(
     },
   )
 }
+
+export function useDeleteOAuthClient(
+  options?: UseMutationOptions<void, SynapseClientError, string>,
+) {
+  const queryClient = useQueryClient()
+  const { accessToken } = useSynapseContext()
+
+  return useMutation<void, SynapseClientError, string>(
+    (clientId: string) =>
+      SynapseClient.deleteOAuthClient(clientId, accessToken!),
+    {
+      ...options,
+      onSuccess: async (updatedClient, clientId, ctx) => {
+        await queryClient.invalidateQueries(oAuthQueryKeys.all(accessToken!))
+        queryClient.setQueryData(
+          oAuthQueryKeys.edit(accessToken!, clientId),
+          updatedClient,
+        )
+        if (options?.onSuccess) {
+          await options.onSuccess(updatedClient, clientId, ctx)
+        }
+      },
+    },
+  )
+}
+
+export function useMutateOAuthClient(
+  options?: UseMutationOptions<OAuthClient, SynapseClientError, OAuthClient>,
+) {
+  const queryClient = useQueryClient()
+  const { accessToken } = useSynapseContext()
+
+  const update = (props: {
+    client: OAuthClient
+    action: 'UPDATE' | 'CREATE'
+  }): Promise<OAuthClient> => {
+    const { client, action } = props
+    switch (action) {
+      case 'UPDATE':
+        return SynapseClient.updateOAuthClient(client, accessToken!)
+      case 'CREATE':
+        return SynapseClient.createOAuthClient(client, accessToken!)
+      default:
+        throw Error('Unknown action')
+    }
+  }
+
+  return useMutation(update, {
+    onSuccess: async (updatedClient, variables, ctx) => {
+      const clientId = updatedClient.client_id
+      queryClient.invalidateQueries(oAuthQueryKeys.all(accessToken!))
+      queryClient.setQueryData(
+        oAuthQueryKeys.edit(accessToken!, clientId!),
+        updatedClient,
+      )
+      if (options?.onSuccess) {
+        await options.onSuccess(updatedClient, variables.client, ctx)
+      }
+    },
+    onError: (err: any) => {
+      displayToast(err.reason as string, 'danger')
+    },
```

Comment:
This level should just manage data. Let the caller decide how to present errors.
```suggestion
```